plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,163 @@
+local util = require("ffi/util")
+local _ = require("gettext")
+local Screen = require("device").screen
+
+
+local TextEditor = WidgetContainer:new{
+    name = "text_editor",
+}
+
+function TextEditor:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function TextEditor:start(file_path)
+    local FileManager = require("apps/filemanager/filemanager")
+    FileManager:onClose()
+    if not file_path then
+        self.context = ""
+        self.file_path = ""
+        self:createUI()
+    else
+        self.file_path = file_path
+        self:createUI()
+        self:selectFileForOpen(file_path)
+    end
+end
+
+function TextEditor:createUI()
+    self.input = InputDialog:new{
+        title =  _("Basic text editor"),
+        input_type = "string",
+        input = self.context,
+        fullscreen = true,
+        allow_newline = true,
+        cursor_at_end = false,
+        add_scroll_buttons = true,
+        add_nav_bar = true,
+        buttons = {{
+        {
+            text = _("Open"),
+            callback = function()
+                self:chooseFile()
+            end,
+        },
+        {
+            text = _("Save"),
+            callback = function()
+                self:saveFile(self.file_path)
+            end,
+        },
+        {
+            text = _("Close"),
+            callback = function()
+                UIManager:show(ConfirmBox:new{
+                    text = _("Are you sure that you want to close the editor? All unsaved changes will be lost."),
+                    ok_text = _("Close"),
+                    ok_callback = function()
+                        self.context = ""
+                        self.file_path = ""
+                        UIManager:close(self.input)
+                    end
+                })
+
+            end,
+        },
+
+    }},
+    }
+    UIManager:show(self.input)
+    self.input:onShowKeyboard()
+end
+
+function TextEditor:chooseFile()
+    self.input:onClose()
+    local path_chooser = PathChooser:new{
+        title = _("Open file. Long press to confirm"),
+        height = Screen:getHeight(),
+        path = util.realpath(DataStorage:getDataDir()),
+        show_hidden = G_reader_settings:readSetting("show_hidden"),
+        file_filter = function() return true end,
+        onConfirm = function(file_path)
+            logger.dbg("TextEditor: selected file_path " .. file_path )
+            self.file_path = file_path
+            self:selectFileForOpen(self.file_path)
+            self.input:onShowKeyboard()
+        end
+    }
+    UIManager:show(path_chooser)
+end
+
+function TextEditor:selectFileForOpen(file_path)
+    logger.dbg("TextEditor: reading file: " .. file_path)
+    local file = io.open(file_path, "rb")
+    if file then
+        local start_file = file:seek()
+        if file:seek("end") > 32000 then
+            UIManager:show(ConfirmBox:new{
+                text = _("This seems a binary or very big file. Are you sure you want to open it?"),
```

Comment:
We have the info, so I think it's more useful to give it.

```
This file is %1. Are you sure you want to open it?

Reply:
> This file is 33957. Are you sure you want to open it?

Are you sure it's useful info? 
Personally I'd prefer info that it might be too big for simple plugin.

Reply:
Not in that form... the utility functions are there for a reason.

But yes, to me a generic message is significabtly unhelpful.

Reply:
> the utility functions are there for a reason.

Do you want me to use a utility function? Cause it's not clear for me

> But yes, to me a generic message is significabtly unhelpful

Before I change it (as I'm not a fan of this, as we have the same info during opening in "path chooser" ) can we have at least one more vote here? 
Can some else say do you prefer info that's "too big " or just info saying "size is "?

Reply:
There is some `util.getFriendlySize()`, so you could have
`The file is 123KB. Are you sure you want to open it?\n(opening big files may take time).`

I'd be more sure I want to open it if I know it's only 123KB.
And not so sure if this tells me it is 12,5MB :)